javascript/organizing_your_javascript_code/organizing_code_with_objects.md

```diff
@@ -0,0 +1,374 @@
+- A "Debt" object that keeps track of a debt owed, how much has been paid, and how much of the debt is remaining.
+
+You may have trouble figuring out what kinds of properties and methods one of these objects might contain at first. One way you might conceptualize these objects might be to imagine them as little 'machines' you're making out of code.
+
+The properties of the machine could be thought of displays that might show information such as:
+
+- A list of the items you've collected, the total amount of items you can carry, and how much you're currently carrying
+- A list of functions that are listening for an event
+- The person who owes a particular debt, and the original amount owed.
+
+The methods of your machine might be akin to buttons and such that make the machinde *do* a specific thing, such as:
+
+- Remove an item you own from a list, add a new item, upgrade an item, craft a new item
+- Fire all the functions that are listening to a 'click' event, or add a new function to listen to the 'click' event
+- Pay an amount of money towards a debt, and determine how much more money is owed on a debt
+
+Again, objects can be used to represent almost anything you can think of, the limit is your imagination!
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+1. Check out this [video explanation](https://www.youtube.com/watch?v=cwChC4BQF0Q) on the `this` keyword from DevSage that gives a different perspective on how its context changes, as well as scenarios in which `this` behaves unexpectedly. Don't worry too much about the part on constructor functions at the end, as they will be covered in another lesson.
+
+1. Try modelling the behavior of a piggy bank as an object:
+   1. You should be able to interact with the object via a method that allows you to deposit coins.
+      - For now, use a string to represent the name of a coin. Ex. `piggyBank.deposit('penny')`
+      - You should keep the coin around by storing it in some kind of list.
+   1. The object should also have a property or method that tells you how much money you have saved as a number.
+      - For simplicity, using only the smallest subunit of a given currency is acceptable.
+      - Ex: If there are 5 US quarters in the piggy bank, then the number `125` for '125 cents' is acceptable.
+   1. You should be able to read the list of all coins that are currently in the piggy bank.
+   1. Add a method to the piggy bank object to remove a specific type of coin from the piggy bank if it's available.
+      - Ensure you can still get the correct savings after removing coins!
+   1. Lastly, create a way to remove all the money from the jar and start fresh.
+
+1. Try to model something else as an object! Try to keep it simple. When you're done, consider showing off what you've created on [The Odin Project's Discord Channel](https://discord.com/invite/fbFCkYabZB), and tell us how you're feeling about the idea of organizing code into objects.
```

Comment:
I'm not quite so sure this is necessary as a assignment step but neither do I feel particularly strongly about it. Below is tweaked wording for the link and also using the appropriate href for the creations showcase channel. The server main invite link has been linked many times before this lesson already so I feel it's fine to link straight to a channel here.

```suggestion
1. Try to model something else as an object! Try to keep it simple. When you're done, consider showing off what you've created in [our Discord server's "Creations Showcase" channel](https://discord.com/channels/505093832157691914/1013800461465309214).
```

Reply:
I noticed there was something like this in a couple lessons, and there was also discussion about discord engagement for a little bit after a fireside chat type thing. 

I believe I asked about it somewhere, and thoughts on it were inconclusive. I don't feel particularly strongly about it either, but it seems a little fun. If you're erring on the side of leaving it out, tell me and it'll get wiped.

Reply:
I'm probably erring on the side of "leave it out" for now. Again, I don't feel too strongly about its inclusion/exclusion.